src/Microsoft.Extensions.PlatformAbstractions/Native/PlatformApis.cs

```diff
@@ -0,0 +1,164 @@
+                return "10.0";
+            }
+            else
+            {
+                // Mac OS X 10.1 mapped to Darwin 5.x, and the mapping continues that way
+                // So just subtract for from the Darwin version.
+                // https://en.wikipedia.org/wiki/Darwin_%28operating_system%29
+                return $"10.{version.Major - 4}";
+            }
+        }
+
+        public static Platform GetOSPlatform()
+        {
+            return _platform.Value;
+        }
+
+        private static string GetDistroId()
+        {
+            return _distroInfo.Value?.Id;
+        }
+
+        private static string GetDistroVersionId()
+        {
+            return _distroInfo.Value?.VersionId;
+        }
+
+        private static DistroInfo LoadDistroInfo()
+        {
+            // Sample os-release file:
+            //   NAME="Ubuntu"
+            //   VERSION = "14.04.3 LTS, Trusty Tahr"
+            //   ID = ubuntu
+            //   ID_LIKE = debian
+            //   PRETTY_NAME = "Ubuntu 14.04.3 LTS"
+            //   VERSION_ID = "14.04"
+            //   HOME_URL = "http://www.ubuntu.com/"
+            //   SUPPORT_URL = "http://help.ubuntu.com/"
+            //   BUG_REPORT_URL = "http://bugs.launchpad.net/ubuntu/"
+            // We use ID and VERSION_ID
+
+            if (File.Exists("/etc/os-release"))
+            {
+                var lines = File.ReadAllLines("/etc/os-release");
+                var result = new DistroInfo();
+                foreach (var line in lines)
+                {
+                    if (line.StartsWith("ID=", StringComparison.Ordinal))
+                    {
+                        result.Id = line.Substring(3).Trim('"', '\'');
+                    }
+                    else if (line.StartsWith("VERSION_ID=", StringComparison.Ordinal))
+                    {
+                        result.VersionId = line.Substring(11).Trim('"', '\'');
+                    }
+                }
+                return result;
+            }
+            return null;
+        }
+
+        private static Platform DetermineOSPlatform()
+        {
+#if NET451
+            var platform = (int)Environment.OSVersion.Platform;
+            var isWindows = (platform != 4) && (platform != 6) && (platform != 128);
+
+            if (isWindows)
+            {
+                return Platform.Windows;
+            }
+            else
+            {
+                try
+                {
+                    var uname = NativeMethods.Unix.GetUname();
+                    if (string.Equals(uname, "Darwin", StringComparison.OrdinalIgnoreCase))
+                    {
+                        return Platform.Darwin;
+                    }
+                    if (string.Equals(uname, "Linux", StringComparison.OrdinalIgnoreCase))
+                    {
+                        return Platform.Linux;
+                    }
+                    return Platform.Unknown;
+                }
+                catch
```

Comment:
I was going to make some joke about how you told me in person there wasn't really anything in the PR feedback that I had to do so I wasn't going to do it because "YOU CAN'T CONTROL ME"...

But I agree... so I'm going to do it.